packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -723,9 +729,16 @@ export class DeltaManager
         // const serializedContent = JSON.stringify(this.messageBuffer);
         // const maxOpSize = this.context.deltaManager.maxMessageSize;
 
-        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-        if (this.readonly) {
-            this.close(CreateContainerError("Op is sent in read-only document state"));
+        if (this.readonly === true) {
+            assert(this.readOnlyInfo.readonly === true, "Unexpected mismatch in readonly");
```

Comment:
i wouldn't assert, we throw next anyway, and then we will not get the telemtry

Reply:
Chatted with Tony offline -- this is needed because `ReadOnlyInfo` is a discriminated union so Typescript disallows property access for `forced`/`permissions`/`storageOnly` without first verifying `readOnlyInfo.readonly` is true.  By current definition of `deltaManager.readonly` this assert should always pass though since `readOnlyInfo.readonly` returns true in a strict superset of instances when `deltaManager.readOnly` returns true - and if that changes it's probably good for us to assert and break here.